src/rfc/detail/query_part_rule.hpp

```diff
@@ -50,11 +50,11 @@ struct query_part_rule_t
         ++it;
         auto rv = grammar::parse(
             it, end, query_rule);
-        if(! rv)
-            return rv.error();
+        BOOST_ASSERT_MSG( rv, "query_rule is optionally empty and must not fail" );
```

Comment:
when did we start putting strings in the asserts, instead of just leaving a comment before the line?